invenio_users_resources/services/users/service.py

```diff
@@ -36,18 +46,18 @@ def user_cls(self):
 
     @unit_of_work()
     def create(self, identity, data, raise_errors=True, uow=None):
-        """Create a user."""
+        """Create a user from users admin."""
         self.require_permission(identity, "create")
-
-        # validate data
+        # Remove None values to avoid validation issues
+        data = {k: v for k, v in data.items() if v}
+        # validate new user data
         data, errors = self.schema.load(
             data,
             context={"identity": identity},
+            raise_errors=raise_errors,
         )
-
-        # create the user with the specified data
-        user = self.user_cls.create(data)
-
+        # create user
+        user = self._create_user_as_admin(data)
```

Comment:
why is `admin` in the name? if only admins should be able to do it, it should be handled by the permissions setting, otherwise it feels wrong.